kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -408,42 +429,46 @@ define([
             if (!filePathParams.layout.length) {
                 ui.getElement(`${cssClassType}s-area`).classList.add('hidden');
             } else {
-
                 filePathRow.innerHTML = [
                     td({
                         class: `${cssBaseClass}__file_number`,
                     }),
                     td(
-                        div({
-                            class: `${cssBaseClass}__param_container row`,
-                        }, [
-                            filePathParams.content
-                        ])
+                        div(
+                            {
+                                class: `${cssBaseClass}__param_container row`,
+                            },
+                            [filePathParams.content]
+                        )
                     ),
                     td({}, [
-                        button({
-                            class: `${cssBaseClass}__button--delete btn btn__text`,
-                            type: 'button',
-                            id: events.addEvent({
-                                type: 'click',
-                                handler: function(e){
-                                    deleteRow(e);
-                                }
-                            })
-                        },[
-                            icon({
-                                class: 'fa fa-trash-o fa-lg',
-                            })
-                        ])
-                    ])
+                        button(
+                            {
+                                class: `${cssBaseClass}__button--delete btn btn__text`,
+                                type: 'button',
+                                id: events.addEvent({
+                                    type: 'click',
+                                    handler: function (e) {
+                                        deleteRow(e);
+                                    },
+                                }),
+                            },
+                            [
+                                icon({
+                                    class: 'fa fa-trash-o fa-lg',
+                                }),
+                            ]
+                        ),
+                    ]),
                 ].join('');
 
-                Promise.all(filePathParams.layout.map((parameterId) => {
-                    createFilePathWidget(appSpec, filePathParams, parameterId);
-                })).then(
-                    events.attachEvents(container)
-                );
-
+                Promise.all(
+                    filePathParams.layout.map(async (parameterId) => {
```

Comment:
updated here to be an async await so we can return from promise.all once all widgets have been built